syllabus/curricula/plr/task_sampler.py

```diff
@@ -285,16 +377,64 @@ def sample(self, strategy=None):
                 if np.random.rand() > self.nu or not proportion_seen < 1.0:
                     return self._sample_replay_level()
 
-            # Otherwise, sample a new level
-            return self._sample_unseen_level()
+            # Otherwise, evaluate a new level
+            if self.robust_plr:
+                self.update_with_episode_data(self._evaluate_unseen_level())
+                return self.sample(strategy=strategy)
+            else:
+                # Otherwise, sample a new level
+                return self._sample_unseen_level()
+
 
         elif self.replay_schedule == "proportionate":
             if proportion_seen >= self.rho and np.random.rand() < proportion_seen:
                 return self._sample_replay_level()
             else:
-                return self._sample_unseen_level()
+                if self.robust_plr:
+                    return self._evaluate_unseen_level()
+                else:
+                    return self._sample_unseen_level()
+
         else:
-            raise NotImplementedError(f"Unsupported replay schedule: {self.replay_schedule}. Must be 'fixed' or 'proportionate'.")
+            raise NotImplementedError(
+                f"Unsupported replay schedule: {self.replay_schedule}. Must be 'fixed' or 'proportionate'.")
+
+    def update_with_episode_data(self, episode_data, score_function):
```

Comment:
Make sure whenever you call this function, you're passing in the score_function. It looks like you're passing in different values each time you call it.